src/rep/mesh.jl

```diff
@@ -131,8 +131,8 @@ function TriMesh(
     verts = [T.(v) for v in verts]
     faces = [R.(f) for f in faces]
 
-    _verts_len = S(size.(verts, 2))
-    _faces_len = S(size.(faces, 2))
+    _verts_len = cpu(size.(verts, 2))
```

Comment:
Why do you need to call `cpu` here? Shouldn't the broadcast produce a cpu vector always?

Reply:
my bad. `_verts_len` is always on cpu.